q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,312 @@
+    """Check that the user-defined value for the
+    rule "ontology" is correctly formatted.
+
+    ontology must be a str
+        Name of the ontology to lookup within for control vocabulary, e.g.
+                    ontology: Gazetteer
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "ontology".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    if isinstance(rule_value, str):
+        if rule_value in ['Gazetteer ontology']:
+            return True
+    return False
+
+
+def check_remap(rule_value):
+    """Check that the user-defined value for the
+    rule "remap" is correctly formatted.
+
+    remap must be a dict
+        Apply the rule of remapping variables values: replace
+        all instances of a given value by another one value.
+                    remap:
+                      str: str
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "remap".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+
+    """
+    if isinstance(rule_value, dict):
+        str_values = [x for x, y in rule_value.items() if isinstance(y, str)]
+        if len(str_values) == len(rule_value.keys()):
+            return True
+    return False
+
+
+def check_validation(rule_value):
+    """Check that the user-defined value for the
+    rule "validation" is correctly formatted.
+
+    validation must be a dict
+        Apply the rule of validating variables values: replace by the blank
+        (or the missing) value every sample entry that for the current variable has
+        in one or multiple other variable(s) a null or missing value for that sample.
+                    validation:
+                      force_to_blank_if:
+                        is null:
+                        - variable
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "validation".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+
+    """
+    # accepted structure for the "validation" rule
+    possible_validations = {
+        'force_to_blank_if': {'is null'}
+    }
+    # check the first level is a dict
+    if isinstance(rule_value, dict):
+        # get the current keys
+        rule_value_set = set(rule_value)
+        # get the accepted keys from the current keys
+        accepted_set = rule_value_set & set(possible_validations)
+        # if there is not more keys than the accepted ones
+        if not len(rule_value_set ^ accepted_set):
+            # get the possible conditions
+            possible_validations_2 = possible_validations['force_to_blank_if']
+            # for each of the condition/replacements for the accepted keys
+            for rule_2, rule_value_2 in rule_value['force_to_blank_if'].items():
+                # check that the content of replacements are lists
+                if isinstance(rule_value_2, list):
```

Comment:
Check the contents of the innermost list to get intersection of list contents with metadata variables.

Also, this method could be rewritten recursively maybe?

Reply:
Yes - please @HuRebecca  could you propose a recursive solution?

Reply:
@ElDeveloper - Rebecca and I figured that a rewriting proposition for this function could be submitted later as an amendment to the branch. Hence I would suggest to merge the branch without it if everything is okay for @wasade too.